Lesson31/Kadir/my-app/src/components/AddButton/index.jsx

```diff
@@ -0,0 +1,43 @@
+import React, { useState } from "react";
+import "./styles.css";
+
+export const AddButton = ({ key, data, setData}) => {
+  const [text, setText] = useState("");
+  const [newItem, setNewItem] = useState(); // Define a state to store the newItem
+
+  const handleChange = (event) => {
+    setText(event.target.value);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    text.trim() !== "" &&
+      setNewItem({
+        key: key + 1,
+        id: Math.floor(Math.random() * 1000),
```

Comment:
Normally, we use [UUID](https://www.techtarget.com/searchapparchitecture/definition/UUID-Universal-Unique-Identifier#:~:text=A%20UUID%20(Universal%20Unique%20Identifier,UUID%20generated%20until%20A.D.%203400.) that should be generated in backend however there's a trick for situations like this: Date.now() would generate a current timestamp so you can rely on the value to be unique.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/now

`Math.floor(Math.random() * 1000)` si not bad but is more complex than `Date.now()`.